library/src/main/java/es/voghdev/pdfviewpager/library/remote/DownloadFile.java

```diff
@@ -15,9 +15,13 @@
  */
 package es.voghdev.pdfviewpager.library.remote;
 
+import java.util.Map;
+
 public interface DownloadFile {
     void download(String url, String destinationPath);
 
+    void download(String url, String destinationPath, Map<String, String> headers);
```

Comment:
I think that the HTTP headers are an implementation detail. The DownloadFile abstraction should be agnostic of what's happening behind, so I would place the Map<String, String> as constructor parameter of the DownloadFileImpl. In fact, instead of modifying the current one, I would create a new child class of DownloadFile that includes the headers as parameter, so clients can choose one implementation or the other. Let me know your opinion about this

Reply:
I think in the essence you are right, but when doing this I think we should rename the class/method as well, since URL implicates HTTP already and DownloadFileIm should maybe be renamed to HTTTPFileDownloader?

Reply:
@voghDev What is your opinion?

Reply:
I would create another child class that inherits from DownloadFile, that receives the headers via constructor. It could have a descriptive name like `DownloadFileWithHeadersImpl` or `DownloadFileHTTPHeadersImpl`, so the implementation details (headers in this case), are only relevant for this child class, and not for the original abstraction.

Don't know if that's what you were asking or if it's helpful :-P